modules/shared/atoms/Button/IButton.d.ts

```diff
@@ -0,0 +1,13 @@
+declare namespace IButton {
+  export interface IProps {
+    children?: string;
+    disabled?: false | true;
+    variant?: 'primary' | 'secondary' | 'text';
```

Comment:
is there a reason not to use Enum or another Interface to be implemented instead of adding the options manually?

also using the type "boolean" instead of listing options true false, could make more sense?
e.g:
enum buttonVariantValues{
PRIMARY = "primary"
SECONDARY = "secondary"
TEXT = "text"
}
interface Iprops{
children?: string;
variant: buttonVariantValues;
disabled: boolean
}